olive/engine/engine.py

```diff
@@ -289,7 +289,7 @@ def run(
                     result_name = f"{prefix_output_name}metrics"
                     results_path = output_dir / f"{result_name}.json"
                     with open(results_path, "w") as f:
-                        f.write(results.json())
+                        json.dump(results.to_json(), f, indent=4)
```

Comment:
>to_json [](http://example.com/codeflow?start=42&length=7)

what's the difference between to_json and json()?

Reply:
the .json() returns a json string (it is a built-in method of pydantic BaseModel). `to_json` is a method of `ConfigBase` that returns a json as python dictionary.

Reply:
could I suppose the json.load(json()) == to_json()?

Reply:
Yes, it only pretty prints the json file. the contents are the same.